### DISCUSSION TITLE
Community-Maintained Build Configuration Fork

### DISCUSSION THREAD

**Discussion by preist-nvidia** [NVIDIA-Omniverse/PhysX#58] [Ideas]
Hi all,

We cannot merge most of the build configuration PRs that are coming in because we cannot test and maintain them. So I would suggest the following solution:

Someone sets up a community-maintained build configuration fork of this repo, and we link to it in our readme. @moraaar, for example?

Please let us know what you think.

Thank you,
Philipp

**Comment by Zeblote** [comment]
It would be useful if this fork also fixed the situation where part of the build scripts are downloaded as a separate package rather than part of the repository for seemingly no reason.

**Reply by preist-nvidia** [reply]
Right, you are referring to https://github.com/NVIDIA-Omniverse/PhysX/discussions/48#discussion-4602454.

@ayoub-belarbi for more input.

**Reply by moraaar** [reply]
Thanks @preist-nvidia I've created 2 new issues requesting changes to internal packman packages:
- https://github.com/NVIDIA-Omniverse/PhysX/issues/59
- https://github.com/NVIDIA-Omniverse/PhysX/issues/60

**Comment by moraaar** [comment]
Hi @preist-nvidia, thanks for letting us know your decision with respect to [Mac](https://github.com/NVIDIA-Omniverse/PhysX/pull/51), [iOS](https://github.com/NVIDIA-Omniverse/PhysX/pull/49) and [Android](https://github.com/NVIDIA-Omniverse/PhysX/pull/40) platforms.

I'll share your proposal, discuss it internally and we'll get back to you by replying to this discussion.

If in the case we decide to create this new repo for these platforms it'd be very useful if there would be unit tests or any kind of tests to run in order to verify the platforms still work. I don't think I've seen unit tests as part of the PhysX 5 repo, do you have unit tests internally? If yes, would it be possible to make them part of the repo with instructions of how to run them?

Thanks,
Aaron Ruiz Mora

**Reply by preist-nvidia** [reply]
Thank you for checking this in your team, Aaron! Looking forward to hearing back from you.

Adding our UT suite to the public repository is not possible, unfortunately.

Thank you,
Philipp

**Reply by moraaar** [reply]
Hi @preist-nvidia,

Great news! We just got approval to host a PhysX repo where we can put the macOS/iOS/Android platforms.

> Someone sets up a community-maintained build configuration fork of this repo, and we link to it in our readme. @moraaar, for example?

We thought that it wasn't a good idea that this new fork would be maintained under an individual account, because there is always the risk that the user could close or abandon it. So instead we thought it'd be better for the O3DE organization to host this PhysX fork. This way there are already processes to have maintainers and reviewers by anybody from the community.

**We got approval from O3DE and here is the repo, you can link to it in your readme: https://github.com/o3de/PhysX
This fork already has the macOS, iOS and Android platforms merged into the default branch.**

This fork is of course public for anybody to use. The O3DE users from sig-simulation (me and other collaborators from different companies) are maintainers/reviewers.

A few points to consider:
1. It'd great if somebody from NVIDIA could join us in that fork as a maintainer/reviewer. You would be a member of the community and considering you know PhysX SDK better than anybody else your comments and feedback in Pull Requests would be invaluable. Please let me know what you think about this.

2. Please could you find time to address the following two issues in the CMake files that are downloaded from packman. Right now you are the only person who can fix them because they're not part of the repo. It'll make the experience for using other platforms much better:
  - https://github.com/NVIDIA-Omniverse/PhysX/issues/59
  - https://github.com/NVIDIA-Omniverse/PhysX/issues/60

3. The main purpose of this fork is to take changes related with the new platforms. Any other contribution (generic fixes for example) will be advised to go through the official NVIDIA repo.

In future we can figure out a process to validate the new platforms.

Looking forward to hearing from you!

Thanks for you time,
Aaron

**Comment by preist-nvidia** [comment]
Hi Aaron,

Happy New Year, and what a great start it is given these wonderful news! Thank you so much for setting up the fork. I will add a link to your repo to the readme. As for your points:

1. Yes we can definitely help with questions - I signed up for notifications so I can have a look at issues / discussions / triage in the team.
2. Yes, I'll bring this up in the team. @ayoub-belarbi for viz.
3. Yes that makes sense, thank you.

Validation: A simple but manual approach would be to just test that the snippets run ok for now.

Thank you again so much and Happy New Year!
Philipp

**Comment by fabmax** [comment]
Hi,

would it be possible to maintain some kind of list of additional repositories providing PhysX bindings for different languages (either here or in the o3de fork)? I have two repositories, which provide PhysX bindings for [Java](https://github.com/fabmax/physx-jni) and [Javascript/Webassembly](https://github.com/fabmax/physx-js-webidl)

Another option would be to integrate those bindings directly into the community maintained fork but that would mean cluttering up the repo with different languages and build systems, which I think is not desirable.

Thanks
Max
